psa-crypto/src/types/key_derivation.rs

```diff
@@ -32,7 +32,7 @@ pub enum Inputs<'a> {
         /// Secret, used in the "extract" step. This is typically a key of type `Derive` , or the shared secret
         /// resulting from a key agreement, using `Input::KeyAgreement`.
         /// Must be a key or key agreement input if used with `psa_key_derivation_output_key`.
```

Comment:
Is it worth duplicating these structs to enforce this? So we'd have an operation struct for `output_key` and one for `output_bytes` where the `output_key` struct only allows a key or key agreement for the `secret` step?

Reply:
No, I think that's overkill, as I assume most of the code would have to be duplicated too, right?

Reply:
Yes most of it would have to be duplicated.

Reply:
I'd say writing a few lines to check the validity is better than duplicating tens/hundreds to make ensure that statically. If we have a regression test for it that should be enough. Do you think the static checking would bring a big benefit?

Reply:
Yes, you're probably right. If someone wants to use this on an embedded device, they may end up appreciating the code size savings over enforcing correct parameters.

> Do you think the static checking would bring a big benefit?
Not big, no. It's easy enough to just read the documentation! But if there's a cheap enough way of forcing something to be correct, I'd take it (haven't though of a way to enforce this ).